goal_src/engine/gfx/shrub/shrub-work.gc

```diff
@@ -5,3 +5,793 @@
+      )
+     :mscal-tmpl
+     (new 'static 'dma-packet
+      :dma
+      (new 'static 'dma-tag :id (dma-tag-id next))
+      :vif0
+      (new 'static 'vif-tag :imm #x404 :cmd (vif-cmd stcycl))
+      :vif1
+      (new 'static 'vif-tag :imm #xa :cmd (vif-cmd mscal) :msk #x1)
+      )
+     :init-tmpl
+     (new 'static 'dma-packet
+      :dma
+      (new 'static 'dma-tag :qwc #x2 :id (dma-tag-id next))
+      :vif0
+      (new 'static 'vif-tag :imm #x404 :cmd (vif-cmd stcycl))
+      :vif1
+      (new 'static 'vif-tag :imm #x38a :num #x1 :cmd (vif-cmd unpack-v4-32))
+      )
+     :init-data
+     (new 'static 'array uint32 8 #x0 #x0 #x363 #x0 #x0 #x0 #x0 #x1500000c)
+     )
+    )
+   :dma-ref
+   (new 'static 'dma-packet :dma (new 'static 'dma-tag :id (dma-tag-id ref)))
+   :dma-end
+   (new 'static 'dma-packet :dma (new 'static 'dma-tag :id (dma-tag-id end)))
+   :wind-const
+   (new 'static 'vector :x 0.5 :y 100.0 :z 0.0166 :w -1.0)
+   :constants (new 'static 'vector :x 128.0 :y 1.0)
+   :color-constant
+   (new 'static 'vector4w
+    :data
+    (new 'static 'array int32 4 #x47000000 #x47000000 #x47000000 0)
+    )
+   :start-bank
+   (new 'static 'array uint8 20
+    #x0
+    #x1
+    #x1
+    #x1
+    #x1
+    #x1
+    #x1
+    #x1
+    #x1
+    #x1
+    #x1
+    #x0
+    #x0
+    #x0
+    #x0
+    #x0
+    #x0
+    #x0
+    #x0
+    #x0
+    )
+   )
+  )
+
+;; failed to figure out what this is:
+(set! (-> *instance-shrub-work* mscalf-tmpl vif0 imm) 103)
+
+;; failed to figure out what this is:
+(set! (-> *instance-shrub-work* mscalf-ret-tmpl vif0 imm) 103)
+
+;; definition for function upload-generic-shrub
+;; Used lq/sq
+(defun
+  upload-generic-shrub
+  ((arg0 dma-buffer) (arg1 generic-shrub-fragment) (arg2 int) (arg3 int))
+  (let* ((v1-0 arg0)
+         (t0-0 (the-as object (-> v1-0 base)))
+         )
+   (set!
+    (-> (the-as dma-packet t0-0) dma)
+    (new 'static 'dma-tag :qwc #x6 :id (dma-tag-id cnt))
+    )
+   (set!
+    (-> (the-as dma-packet t0-0) vif0)
+    (new 'static 'vif-tag :imm #x404 :cmd (vif-cmd stcycl))
+    )
+   (set!
+    (-> (the-as dma-packet t0-0) vif1)
+    (new 'static 'vif-tag
+     :num #x6
+     :cmd (vif-cmd unpack-v4-32)
+     :imm (shr (shl arg2 54) 54)
+     )
+    )
+   (set! (-> v1-0 base) (&+ (the-as pointer t0-0) 16))
+   )
+  (let* ((v1-1 (the-as matrix (-> arg0 base)))
+         (t3-0 (-> *math-camera* camera-temp))
+         (t0-3 (-> t3-0 vector 0 quad))
+         (t1-4 (-> t3-0 vector 1 quad))
+         (t2-4 (-> t3-0 vector 2 quad))
+         (t3-1 (-> t3-0 vector 3 quad))
+         )
+   (set! (-> v1-1 vector 0 quad) t0-3)
+   (set! (-> v1-1 vector 1 quad) t1-4)
+   (set! (-> v1-1 vector 2 quad) t2-4)
+   (set! (-> v1-1 vector 3 quad) t3-1)
+   )
+  (set! (-> arg0 base) (&+ (-> arg0 base) 64))
+  (let* ((v1-4 arg0)
+         (t0-4 (the-as object (-> v1-4 base)))
+         )
+   (set! (-> (the-as vector4w-2 t0-4) vector 0 x) #x3f800000)
```

Comment:
These values look like floats to me.  This is fine for now though, I don't really know what type should go here.